src/DependencyInjection/Admin/AbstractTaggedAdmin.php

```diff
@@ -190,16 +207,82 @@ abstract class AbstractTaggedAdmin implements TaggedAdminInterface
     private $templateRegistry;
 
     /**
-     * @phpstan-param class-string<T> $class
+     * NEXT_MAJOR: Remove the __construct method.
+     *
+     * @phpstan-param class-string<T>|null $class
      */
-    public function __construct(string $code, string $class, string $baseControllerName)
+    public function __construct(?string $code = null, ?string $class = null, ?string $baseControllerName = null)
+    {
+        if (\func_num_args() > 0) {
+            @trigger_error(
+                'Setting the code, the model class and the base controller name with the constructor is deprecated'
+                .' since sonata-project/admin-bundle version 4.x and will not be possible in 5.0 version.'
+                .' Use the `code`, `model_class` and `controller` attribute of the `sonata.admin` tag or'
+                .' the method "setCode()", "setModelClass()" and "setBaseControllerName()" instead.',
+                \E_USER_DEPRECATED
+            );
+        }
+
+        if (null !== $code) {
+            $this->code = $code;
+        }
+        $this->class = $class;
+        $this->modelClass = $class;
+
+        if (null !== $baseControllerName) {
+            $this->baseControllerName = $baseControllerName;
+        }
+    }
+
+    abstract public function initialize(): void;
```

Comment:
Can we use this method to initialize this class with all old `__constructor` arguments?